web-console/src/utils/ingestion-spec.tsx

```diff
@@ -1173,11 +1174,35 @@ export function getIoConfigFormFields(ingestionComboType: IngestionComboType): F
           type: 'string-array',
           placeholder: 's3://your-bucket/some-path1, s3://your-bucket/some-path2',
           required: true,
-          defined: ioConfig => !deepGet(ioConfig, 'inputSource.uris'),
+          defined: ioConfig =>
+            !deepGet(ioConfig, 'inputSource.uris') && !deepGet(ioConfig, 'inputSource.objects'),
           info: (
             <>
               <p>A list of paths (with bucket) where your files are stored.</p>
-              <p>Either S3 URIs or S3 prefixes must be set.</p>
+              <p>Either S3 URIs or prefixes or objects must be set.</p>
+            </>
+          ),
+        },
+        {
+          name: 'inputSource.objects',
+          label: 'S3 objects',
+          type: 'json',
+          placeholder: '{"bucket":"your-bucket", "path":"some-file.ext"}',
+          required: true,
+          defined: ioConfig =>
```

Comment:
try defining the `objects` only if they exist, that way they could be more hidden